modules/client-handler/src/main/java/org/apache/ignite/client/handler/requests/jdbc/JdbcMetadataCatalog.java

```diff
@@ -165,19 +229,29 @@ public CompletableFuture<Collection<JdbcColumnMeta>> getColumnsMeta(String schem
         String tlbNameRegex = translateSqlWildcardsToRegex(tblNamePtrn);
         String colNameRegex = translateSqlWildcardsToRegex(colNamePtrn);
 
-        return tablesAtNow().thenApply(tablesList -> tablesList.stream()
-                .filter(t -> tableNameAndSchemaMatches(t, schemaNameRegex, tlbNameRegex))
-                .flatMap(
-                    tbl -> {
-                        SchemaDescriptor schema = CatalogToSchemaDescriptorConverter.convert(tbl, tbl.tableVersion());
-
-                        return schema.columns().stream()
-                                .map(column -> new Pair<>(tbl.name(), column));
-                    })
-                .filter(e -> matches(e.getSecond().name(), colNameRegex))
-                .sorted(bySchemaThenTabNameThenColOrder)
-                .map(pair -> createColumnMeta(pair.getFirst(), pair.getSecond()))
-                .collect(toCollection(LinkedHashSet::new)));
+        return schemasAtNow().thenApply(schemas ->
+                schemas.stream()
+                        .filter(schema -> matches(schema.name(), schemaNameRegex))
+                        .flatMap(schema ->
+                                Stream.concat(
+                                        Arrays.stream(schema.systemViews())
```

Comment:
```suggestion
               stream.concat( Arrays.stream(schema.systemViews())
                                             .map(o -> new Pair(o, o.columns())),
                                        Arrays.stream(schema.tables())
                                            .map(o -> new Pair(o, o.columns()))
                                            )
                                            .filter(p -> matches(p.get1().name(), tlbNameRegex))
                                                .flatMap(p -> p.get2().stream()
```

Reply:
I still don't like using Pair class just because of lack classes hierarchy.
It is better to introduce marker interface for that purposes
```
interface CatalogColumnContainer<T> {
    String name();
    List<T> columns();
}
```
This will make streams code clear.

Reply:
Done, thanks